ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -261,13 +264,28 @@ private void updateTransactionResultTracking(
           transactionSelectionResult.getFrontierCumulativeGasUsed() + gasUsedByTransaction;
     }
 
+    // if it is a GoQuorum private tx, we need to hand craft the receipt
```

Comment:
having another look at it I think we should move this up into the evaluateTransaction method and instead of executing the transaction we should check the validity of the private transaction (see GoQuorumBlockProcessor) and then create the result that we need (publicResultForWhenWeHaveAPrivateTransaction()).
If the private transaction is not valid we have to do the right thing (keep in pool or delete from pool).

Reply:
updated